R/plot_dot.R

```diff
@@ -1159,6 +1166,43 @@ get_value_dependent_sm_edge_style <- function(value, theme){
 # ___________________  ----
 # 2. Measurement Model ----------------------
 
+
+
+#' Tests whether the i_th construct is endogenous or not
+#'
+#' @param model the model object
+#' @param index the index of the construct to test
+#'
+#' @return whether the construct is endogenous or not
+#' @export
+is_sink <- function(model, index) {
```

Comment:
Take a look at whether the `all_endogenous` function defined in inspect_smMatrix.R could be of use here?

Reply:
Sink is for those that are right-most in visualization.. So not all endogenous but only those with no "follow-ups"

Reply:
So sinks are the set of all_endogenous that are not also in all_exogenous? If both those functions exist, it might be as easy as taking a set difference. No probs if you wish to keep it as is though!